python/ray/tune/tests/test_tuner_restore.py

```diff
@@ -301,6 +302,36 @@ def test_tuner_resume_errored_only(ray_start_2_cpus, tmpdir):
     assert sorted([r.metrics.get("it", 0) for r in results]) == sorted([2, 1, 3, 0])
 
 
+def test_tuner_restore_from_cloud(ray_start_2_cpus, tmpdir):
+    """Check that restoring Tuner() objects from cloud storage works"""
+    tuner = Tuner(
+        lambda config: 1,
+        run_config=RunConfig(
+            name="exp_dir",
+            local_dir=str(tmpdir / "ray_results"),
+            sync_config=tune.SyncConfig(upload_dir="memory:///test/restore"),
+        ),
+    )
+    tuner.fit()
+
+    check_path = tmpdir / "check_save"
+    download_from_uri("memory:///test/restore", str(check_path))
+    remote_contents = os.listdir(check_path / "exp_dir")
+
+    assert "tuner.pkl" in remote_contents
+    assert "trainable.pkl" in remote_contents
+
+    (tmpdir / "ray_results").remove(ignore_errors=True)
+
+    tuner2 = Tuner.restore("memory:///test/restore/exp_dir")
```

Comment:
will the new experiment still go to the same cloud directory?

Reply:
It should - the sync config is fetched from the tuner state and will be used to upload again. Let me see if this can be added to the test

Reply:
I've added to the test an assertion that the same cloud dir is updated with a new checkpoint after restoring and calling fit